proposals/lip-Introduce_PoA_module.md

```diff
@@ -0,0 +1,607 @@
+
+*   Rules for `trs.params.threshold` property:
+    *   The value of `trs.params.threshold` is within the following range:
+        *   Minimum value: ⌊ ⅓ × `totalWeight`⌋+ 1
+        *   Maximum value: `totalWeight`
+
+       where ⌊⋅⌋ is the floor function.
+
+*   Rules for `trs.params.validatorsUpdateNonce` property:
+    *   The value of `trs.params.validatorsUpdateNonce` has to be equal to `chainProperties.validatorsUpdateNonce`.
+*   Rules for `trs.params.aggregationBits` and `trs.params.signature` properties:
+    *   The function `verifyWeightedAggSig(keyList, tag, netID, aggregationBits, signature, m, weights, threshold)`, specified in [LIP 0038][BLS], must return VALID where:
+        *   The `keyList` property is an array containing `getValidatorAccount(address).blsKey` for every `address` property in `snapshotStore(0).validators` array sorted in the same order, where `getValidatorAccount` is the function exposed by the [validators module][validatorsModule].  
+        *   The `tag` is equal to `MESSAGE_TAG_POA`.
+        *   The `netID` byte array corresponds to the network ID of the chain.
+        *   The `aggregationBits` argument is the byte array given in `trs.params.aggregationBits`.
+        *   The `signature` argument is the aggregate signature given in `trs.params.signature`.
+        *   The `m` argument is the output bytes of the serialization, as specified in [LIP 0027][lip27], of `trs.params.newValidators`, `trs.params.validatorsUpdateNonce`, and `trs.params.threshold` properties according to the following schema:
+ 
+        ``` java 
+        validatorSignatureMessage = {
+            "type": "object",
+            "properties": {
+               "newValidators: {
+                   "type": "array",
+                   "fieldNumber": 1,    
+                   "items": {
+                       "type": "object",
+                       "required": ["address", "weight"],
+                       "properties": {
+                           "address": {
+                               "dataType": "bytes",
+                               "fieldNumber": 1
+                           }
+                           "weight": {
+                               "dataType": "bytes",
+                               "fieldNumber": 2
+                           }
+                       }
+                   }
+               },
+               "threshold": {
+                   "dataType": "uint64",
+                   "fieldNumber": 2
+               },
+               "validatorsUpdateNonce": {
+                   "dataType": "uint32",
+                   "fieldNumber": 3
+               },
+            },
+            "required": [
+                "newValidators",
+                "threshold",
+                "validatorsUpdateNonce"
+            ]
+        }
+        ```
+        *   The `weights` argument is set to the `weights` property of `snapshotStore(0).validators` array.
```

Comment:
The array itself does not have a `weights` property, but only its elements. Maybe we could say
> The `weights` argument is set to the array containing the `weight` property of the elements in `snapshotStore(0).validators` in the same order.

Or explicit and short

> > The `weights` argument is set to `[ validator.weight for validator in snapshotStore(0).validators]`.

Reply:
thanks, went for the compact suggestion

Reply:
Cbr.